Assets/Scripts/Behaviours/Vehicles/Vehicle_Spawning.cs

```diff
@@ -618,5 +635,150 @@ void SetupDoorsHingeJoints()
+            if (compsUnion.HasExtraOne())
+            {
+                firstExtraIdx = ChooseComponent(Definition.CompRules.nExtraARule, compsUnion.nExtraAComp);
+            }
+
+            if (compsUnion.HasExtraTwo())
+            {
+                firstExtraIdx = ChooseComponent(compsUnion.nExtraBRule, compsUnion.nExtraBComp);
+            }
+
+            foreach (var extra in Extras)
+            {
+                Boolean isFirstExtra = extra.Name == ("extra" + firstExtraIdx.ToString());
+                Boolean isSecondExtra = extra.Name == ("extra" + secondExtraIdx.ToString());
+
+                extra.gameObject.SetActive(isFirstExtra || isSecondExtra);
+            }
+        }
+
+        private    int ChooseComponent(int rule, int comps)
+        {
+            VehicleDef.CompRulesUnion compsUnion = Definition.CompRules;
+
+            if ((rule != 0) && IsValidCompRule(rule))
+            {
+                return ChooseComponentInternal(rule, comps);
+            }
+            else if (UnityEngine.Random.Range(0, 3) < 2)
+            {
+                int[] anVariations = new int[6];
+                int numComps = GetListOfComponentsNotUsedByRules(Extras.Count, anVariations);
+                if (numComps > 0)
+                    return anVariations[UnityEngine.Random.Range(0, numComps)];
+            }
+
+            return -1;
+        }
+
+        private bool IsValidCompRule(int nRule)
+        {
+            //    TODO add weather checking, when weather is implemented.
+            Boolean isRainingNow = false;
+
+            return (nRule != (int)VehicleComponentsRules.ONLY_WHEN_RAINING)
+                || isRainingNow
+            ;
+        }
+
+        private int ChooseComponentInternal(int rule, int comps)
+        {
+            int component = -1;
+            if (rule == (int)VehicleComponentsRules.ALLOW_ALWAYS || 
+                rule == (int)VehicleComponentsRules.ONLY_WHEN_RAINING)
+            {
+                int iNumComps = CountCompsInRule(comps);
+                int rand = UnityEngine.Random.Range(0, iNumComps);
+                component = (comps >> (4 * rand)) & 0xF;
+            }
+            else if (rule == (int)VehicleComponentsRules.MAYBE_HIDE)
+            {
+                int iNumComps = CountCompsInRule(comps);
+                int rand = UnityEngine.Random.Range(-1, iNumComps);
+                if (rand != -1)
+                    component = (comps >> (4 * rand)) & 0xF;
+            }
+            else if (rule == (int)VehicleComponentsRules.FULL_RANDOM)
+            {
+                component = UnityEngine.Random.Range(0, 5);
+            }
+
+            return component;
+        }
+
+        int CountCompsInRule(int comps)
+        {
+            int result = 0;
+            while (comps != 0)
+            {
+                if ((comps & 0xF) != 0xF)
+                    ++result;
+
+                int a = comps / 16;
+                int b = comps >> 4;
```

Comment:
`a` and `b` variables are unused and can be removed

Reply:
Yes, used to check them if I remember how to shift digits properly, and forgot to remove.